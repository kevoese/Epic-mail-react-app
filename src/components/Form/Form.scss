.form {
	flex: 1 1 50%;
	min-width: 400px;
	max-width: 500px;
	display: flex;
	justify-content: center;
	align-items: center;
	flex-flow: column;
	overflow: hidden;
	position: relative;

	 
	.invalid{
		white-space: nowrap;
		color: red;
		font-size: 0.9em;
		text-align: center;
		position: absolute;
		left: 50%;
		margin: 0 auto;
		padding: 0 1.5em;
		transform: translateX(-50%);
		transition: transform 0.5s ease-out;
	}

	.input {
		display: block;
		font-family: cambria;
		font-size: 1em;
		border: none;
		color: rgb(17, 17, 17);
		background-color: transparent;
		border-bottom: 1px solid rgba(39, 39, 39, 0.4);
		width: 22em;
		min-width: 20em;
		padding: 0.5em;
		margin: 2em auto;

		&:hover {
			box-shadow: 0px 4px 7px -5px rgba(0, 0, 0, 0.5);
		}
	}
	::-webkit-input-placeholder {
		color: rgba(11, 11, 15, 0.8);
		font-size: 0.9em;
	}

	.linfo {
		position: relative;
		color: var(--myLightColor);
	}

	.validIcon {
		color: var(--myColor) ;
	}

	.username {
		&::before {
			content: '@epicmail.com';
			position: absolute;
			right: 10%;
			top: 50%;
			transform: translateY(-50%);
			color: #171717 !important;
		}
		input {
			padding-right: 60% !important;
		}
	}

	.showform {
		transform: scale(1);
		opacity: 1;
	}

	h2 {
		color: var(--myColor);
	}

	.mobilebtn {
		margin: 0.5em;
		text-align: center;
		display: none;
		width: 90%;
		justify-content: center;
		align-items: center;
		font-size: 1em;
		font-family: cambria;
		font-style: italic;
		color: var(--myColor);
	}

	.mobilebtn span {
		border: none;
		background-color: transparent;
		font-size: 1em;
		font-family: cambria;
		font-style: italic;
		cursor: pointer;
		color: var(--myColor);
		padding: 5px;
		margin-left: 0.2em;
	}

	.mobilebtn span:hover {
		border-bottom: 1px solid var(--myColor);
	}

	.slider h3 {
		font-size: 2em;
		font-family: myfont;
		color: whitesmoke;
		margin-bottom: 0.5em;
	}

	.slider img {
		width: 10em;
		height: 10em;
		margin: 0 auto;
	}

	.slider h2 {
		font-family: myitalic;
		font-weight: 100;
		color: white;
		flex-wrap: wrap;
		font-size: 4.5em;
		text-align: center;
		margin-bottom: 0.2em;
	}

	.wrongemail {
		input {
			border: 1px solid red;
			animation: wrong 0.4s 0.2s linear backwards;
		}
	}

	.mobilebtn {
		margin: 0.5em;
		text-align: center;
		display: none;
		width: 90%;
		justify-content: center;
		align-items: center;
		font-size: 1em;
		font-family: cambria;
		font-style: italic;
    color: var(--myColor);
    
		span {
			border: none;
			background-color: transparent;
			font-size: 1em;
			font-family: cambria;
			font-style: italic;
			cursor: pointer;
			color: var(--myColor);
			padding: 5px;
			margin-left: 0.2em;

			&:hover {
				border-bottom: 1px solid var(--myColor);
			}
		}
	}
}
