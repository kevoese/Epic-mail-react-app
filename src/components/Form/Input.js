/* eslint-disable no-unused-expressions */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';

class Input extends Component {
  state = {
    valid: false,
  };

  validate = event => {
    event.target.checkValidity() ? this.setState({ valid: true }) : this.setState({ valid: false });
    this.props.onChange(event);
  };

  render() {
    const { id, value, type, pattern, placeholder, title, classes } = this.props;
    return (
      <div className={`icon linfo ${classes || ''} ${this.state.valid && 'validIcon'}`}>
        <input
          id={id}
          value={value}
          onChange={this.validate}
          type={type}
          pattern={pattern}
          className="input"
          placeholder={placeholder}
          required
          title={title}
        />
      </div>
    );
  }
}

export default Input;
