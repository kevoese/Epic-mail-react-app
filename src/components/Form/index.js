import React from 'react';
import './Form.scss';

const Form = ({ children, signup, onSubmit, id, isSubmitting }) => {
  return (
    <form id={id} onSubmit={onSubmit} className={`form ${isSubmitting && ''}`}>
      <div className="formcenter">
        <h2>{signup ? 'Sign up' : 'Sign In'}</h2>
        {/* <div className="mobilebtn">
          <p>{signup ? 'Already a member?' : "Don't have an account?"}</p>
          <span className="mobilelog">{signup ? 'Login here' : 'Register here'}</span>
        </div> */}
        {children}
      </div>
    </form>
  );
};

export default Form;
