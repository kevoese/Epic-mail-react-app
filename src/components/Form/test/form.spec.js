/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import 'jsdom-global/register';
import Form from '@components/Form';
import Input from '@components/Form/input';

describe('Form component', () => {
  it('should render the input component', () => {
    const handleChange = jest.fn();
    const value = '';
    const component = shallow(
      <Input
        id="password"
        value={value}
        onChange={handleChange}
        classes="passwrd"
        pattern="^[\w@.]{7,20}$"
        type="password"
        placeholder="Password"
        required
      />
    );
    expect(component).toMatchSnapshot();
  });
});
