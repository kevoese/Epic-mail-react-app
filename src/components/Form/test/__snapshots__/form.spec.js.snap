// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render the input component 1`] = `
<div
  className="icon linfo passwrd false"
>
  <input
    className="input"
    id="password"
    onChange={[Function]}
    pattern="^[\\\\w@.]{7,20}$"
    placeholder="Password"
    required={true}
    type="password"
    value=""
  />
  <div
    className="e-icon"
  />
</div>
`;
